plugins/by-name/gx/default.nix

```diff
@@ -0,0 +1,117 @@
+  '';
+
+  maintainers = [ lib.maintainers.jolars ];
+
+  extraOptions = {
+    disableNetrwGx = lib.mkOption {
+      default = true;
+      description = "Disable the `gx` mapping in netrw.";
+      type = types.bool;
+    };
+  };
+
+  extraConfig = cfg: {
+    globals = lib.mkIf cfg.disableNetrwGx {
+      netrw_nogx = true;
+    };
+  };
+
+  settingsOptions = {
+    open_browser_app = defaultNullOpts.mkStr null ''
+      Specify your browser app; default for macOS is `"open"`, Linux `"xdg-open"`
+      and Windows `"powershell.exe"`.
+    '';
+
+    open_browser_args = defaultNullOpts.mkListOf types.str [ ] ''
+      Arguments provided to the browser app, such as `--background` for macOS's `open`.
+    '';
+
+    handlers = defaultNullOpts.mkAttrsOf types.anything { } ''
+      Enable built-in handlers and configure custom ones. By default, all
+      handlers are disabled. To enable a built-in handler, set it to `true`.
+    '';
+
+    handler_options = {
+      search_engine = defaultNullOpts.mkStr "google" ''
+        Search engine to use when searching for text. Built-in options are
+        `"google"`, `"duckduckgo"`, `"ecosia"`, `"yandex"`, and `"bing"`. You
+        can also pass a custom search engine, such as
+        `"https://search.brave.com/search?q="`.
+      '';
+
+      select_for_search = defaultNullOpts.mkBool false ''
+        If your cursor is e.g. on a link, the pattern for the link AND for the word
+        will always match. This disables this behaviour for default so that the link is
+        opened without the select option for the word AND link.
+      '';
+
+      git_remotes = defaultNullOpts.mkListOf types.str [ "upstream" "origin" ] ''
+        List of git remotes to search for git issue linking, in priority.
+        You can also pass a function.
+
+        Example:
+        ```nix
+        git_remotes.__raw = \'\'
+          function(fname)
+            if fname:match("myproject") then
+                return { "mygit" }
+            end
+            return { "upstream", "origin" }
+          end
+        \'\';
+        ```
+      '';
+
+      git_remote_push = defaultNullOpts.mkBool false ''
+        Whether to use the push url for git issue linking.
+        You can also pass a function.
+
+        Example:
+        ```nix
+        git_remote_push.__raw = \'\'
+          function(fname) 
+            return fname:match("myproject")
+          end
+        \'\';
+        ```
+      '';
+    };
+  };
+
+  settingsExample = {
+    open_browser_args = [ "--background" ];
+    handlers = {
+      rust = {
+        name = "rust";
+        filetype = [ "toml" ];
+        filename = "Cargo.toml";
+        handle.__raw = ''
+          function(mode, line, _)
+            local crate = require("gx.helper").find(line, mode, "(%w+)%s-=%s")
+
+            if crate then
+              return "https://crates.io/crates/" .. crate
+            end
+          end
+        '';
```

Comment:
I think this will end up rendering poorly in the docs.

Maybe it'd be better to have the settingExample be a literal expression?

```nix
settingsExample = lib.literalExpression ''
  {
    # ...
    handlers = {
      rust = {
        name = "rust";
        filetype = [ "toml" ];
        filename = "Cargo.toml";
        handle.__raw = '''
          function(mode, line, _)
            local crate = require("gx.helper").find(line, mode, "(%w+)%s-=%s")
            if crate then
              return "https://crates.io/crates/" .. crate
            end
          end
        ''';
    # ...
```

Reply:
Thanks, I updated the example. By the way, is there some way to preview the documentation locally? I couldn't find anything about this in the contributing section.

Reply:
Great, thanks for the pointer!